modules/core/shared/src/main/scala/retry/ResultHandler.scala

```diff
@@ -0,0 +1,56 @@
+package retry
+
+import cats.{Applicative, Functor}
+import cats.syntax.functor.*
+
+/** A handler that inspects the result of an action and decides what to do next. This is also a good place to
+  * do any logging.
+  */
+type ResultHandler[F[_], -Res, A] = (Res, RetryDetails) => F[HandlerDecision[F[A]]]
+
+// Type aliases for different flavours of handler
+type ValueHandler[F[_], A]        = ResultHandler[F, A, A]
+type ErrorHandler[F[_], A]        = ResultHandler[F, Throwable, A]
+type ErrorOrValueHandler[F[_], A] = ResultHandler[F, Either[Throwable, A], A]
+
+object ResultHandler:
+  /** Construct an ErrorHandler that always chooses to retry the same action, no matter what the error.
+    *
+    * @param log
+    *   A chance to do logging, increment metrics, etc
+    */
+  def retryOnAllErrors[F[_]: Functor, A](
+      log: (Throwable, RetryDetails) => F[Unit]
```

Comment:
Personal opinion here, but I think `log` is kind of couple to only logging here.

Reply:
Yeah, I couldn't think of a better name